src/client/components/modals/Organization.js

```diff
@@ -0,0 +1,236 @@
+import React, { useContext, useState } from 'react'
+import { Field, Text, Button, theme } from '@aragon/ui'
+import PropTypes from 'prop-types'
+import { ModalWrapper, DisplayErrors } from './ModalWrapper'
+import { validateDAOAddress } from '../../utils/validation'
+import { shortDAOAddress, fakeIsMember } from '../../utils'
+import { BoxContext } from '../../wrappers/box'
+import { ModalContext } from '../../wrappers/modal'
+import { close } from '../../stateManagers/modal'
+import { isEmpty } from 'lodash'
+import styled from 'styled-components'
+import { useProfileStates } from '../../hooks'
+import {
+  requestedCheckMembership,
+  requestedCheckMembershipReset,
+  requestedCheckMembershipSuccess,
+  requestedCheckMembershipError,
+} from '../../stateManagers/box'
+import {
+  TextInputWithValidation,
+  AnimationLoadingCircle,
+  IconSuccess,
+  IconError,
+} from '../../components/styled-components'
+
+const CheckWrapper = styled.div`
+  height: 146px;
+  display: flex;
+  align-items: center;
+  padding-bottom: 13px;
+  > :first-child {
+    margin-left: 10px;
+    margin-right: 30px;
+  }
+`
+
+const BackToPreviousScreen = ({ ethereumAddress, dispatch }) => (
+  <Text.Block
+    css={`
+      text-align: right;
+      margin-bottom: 0px;
+      text-decoration: underline;
+      cursor: pointer;
+      vertical-align: text-bottom;
+      position: absolute;
+      bottom: 0;
+      right: 0;
+    `}
+    color={theme.accent}
+    onClick={() => dispatch(requestedCheckMembershipReset(ethereumAddress))}
+  >
+    Back to previous screen
+  </Text.Block>
+)
+
+BackToPreviousScreen.propTypes = {
+  ethereumAddress: PropTypes.string.isRequired,
+  dispatch: PropTypes.func.isRequired,
+}
+
+const CheckInfo = ({
+  type,
+  ethereumAddress,
+  address,
+  dispatch,
+  linkBack = false,
+}) => {
+  const checkInfoIcon = {
+    started: AnimationLoadingCircle,
+    success: IconSuccess,
+    error: IconError,
+  }
+  const checkInfoText = {
+    started: `Validating your membership to ${shortDAOAddress(address)}...`,
+    success: `We found ${shortDAOAddress(
+      address
+    )} and confirmed you are a member`,
+    error: `We could not verify your membership to ${shortDAOAddress(address)}`,
+  }
+
+  return (
+    <ModalWrapper>
+      <CheckWrapper>
+        {checkInfoIcon[type]()}
```

Comment:
Clever! I didn't know you could render a component by calling a function like this. Neat.

Regarding a point @Schwartz10 originally made:

> on line 84 you can simply do:
> 
>     <CheckInfoIcon { ...type } /> (I think that destructuring should work)
> 
> I believe this will reduce # of renders

Do you know if this new approach you took accomplishes the same "reduce number of renders" goal? 

I doubt it will matter for the performance of this app if it does not, but I'm curious to know.

Reply:
Not sure about the number - I'd need to set up a better testing env than I have here... But I improved this code a bit.